spiketoolkit/postprocessing/postprocessing_tools.py

```diff
@@ -829,41 +847,129 @@ def compute_channel_spiking_activity(recording, channel_ids=None, detect_thresho
                     if len(idx_spikes) > 0:
-                        activity[i] = len(idx_spikes[0])
+                        spike_rates[i] = len(idx_spikes[0]) / duration
+                        spike_amplitudes[i] = np.median(trace[idx_spikes])
                     else:
-                        activity[i] = 0
+                        spike_rates[i] = 0
+                        spike_amplitudes[i] = 0
             else:
                 output_list = Parallel(n_jobs=n_jobs)(
                     delayed(_extract_activity_one_channel)(rec_arg, ch, detect_sign, detect_threshold, start_frame,
                                                            end_frame, verbose)
                     for ch in channel_ids)
 
                 for i, out in enumerate(output_list):
-                    activity[i] = out
-
+                    spike_rates[i] = out[0] / duration
+                    spike_amplitudes[i] = out[1]
         else:
             sort_detect = st.sortingcomponents.detect_spikes(recording, channel_ids=channel_ids,
                                                              detect_threshold=detect_threshold,
                                                              align=False, n_jobs=n_jobs, start_frame=start_frame,
                                                              end_frame=end_frame, verbose=verbose)
             for i, unit in enumerate(sort_detect.get_unit_ids()):
-                n_spikes = len(sort_detect.get_unit_spike_train(unit))
-                activity[i] = n_spikes
-
-        # compute spike rates
-        if start_frame is None:
-            start_frame = 0
-        if end_frame is None:
-            end_frame = recording.get_num_frames()
-        duration = (end_frame - start_frame) / recording.get_sampling_frequency()
-
-        activity /= duration
-        
+                spike_rates[i] = sort_detect.get_unit_property(unit, 'spike_rate')
+                spike_amplitudes[i] = sort_detect.get_unit_property(unit, 'spike_amplitude')
+
         if save_property_or_features:
             for i, ch in enumerate(recording.get_channel_ids()):
-                recording.set_channel_property(ch, 'activity', activity[i])
+                recording.set_channel_property(ch, 'spike_rate', spike_rates[i])
+                recording.set_channel_property(ch, 'spike_amplitude', spike_amplitudes[i])
+
+    return spike_rates, spike_amplitudes
 
-    return activity
+
+def compute_unit_coms(recording, sorting, unit_ids=None, num_channels=10, **kwargs):
+    '''
+    Computes the center of mass (COM) of a unit based on the template amplitudes.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor
+    sorting: SortingExtractor
+        The sorting extractor
+    unit_ids: list
+        List of unit ids to extract maximum channels
+    num_channels: int
+        Number of max channels to be used in the calculation of COM
```

Comment:
Instead of the number of channel I would also add a keyword around the aplitude on each channel in with `k x MAD`
The number of channel is template dependant.

Reply:
we can upgrade it later :)